### PR TITLE
ClrAssembly improvements

### PR DESCRIPTION
**Pull request by jp2masa**
## Changes
- ClrAssembly improvements:
  - Incremental targets.
  - Better support for the dotnet cli.
  - Clean deletes all generated assemblies.
- Fixed Dlr.sln broken paths.

### CODE REVIEW

File: Tests/ClrAssembly/ClrAssembly.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>net40;net45;netcoreapp2.0</TargetFrameworks>
-    <PlatformTarget Condition="'$(TargetFramework)' == 'netcoreapp2.0'">x64</PlatformTarget>
+    <PlatformTarget Condition="'$(TargetFramework)' == 'netcoreapp2.0' AND '$(OS)' != 'Windows_NT'">x64</PlatformTarget>
```
Comment (jp2masa) [inline Tests/ClrAssembly/ClrAssembly.csproj]: Not fully sure about this change, but I think that x86 Windows is supported by .NET Core.

Comment (slide) [reply Tests/ClrAssembly/ClrAssembly.csproj]: The problem is specifying Any CPU, that doesn't work.

Comment (jp2masa) [reply Tests/ClrAssembly/ClrAssembly.csproj]: I can't understand why it doesn't work. It doesn't work on Mono? AnyCPU is the default in the dotnet sdk and it usually works.

Comment (slozier) [reply Tests/ClrAssembly/ClrAssembly.csproj]: Do we even need `PlatformTarget` in `ClrAssembly`? Doesn't it only affect the dummy assembly which we don't use?

**Pull request merged**